android/app/src/main/java/com/zulipmobile/notifications/GCMPushNotifications.java

```diff
@@ -172,12 +180,13 @@ private static void updateNotification(
             builder.addAction(action);
         }
 
+        // Note: This code only set custom sound for API < 26 (from higher API sound is managed by channels)
         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
             AudioAttributes audioAttr = new AudioAttributes.Builder()
                     .setUsage(AudioAttributes.USAGE_NOTIFICATION).build();
-            builder.setSound(Uri.parse("android.resource://" + context.getPackageName() + "/" + R.raw.zulip), audioAttr);
+            builder.setSound(Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE +"://" + context.getPackageName() + NOTIFICATION_SOUND_PATH), audioAttr);
```

Comment:
I agree with using `SCHEME_ANDROID_RESOURCE`, but I don't think the reasons in your commit message are realistic. :-)

The upstream docs [say this constant](https://developer.android.com/reference/android/content/ContentResolver#SCHEME_ANDROID_RESOURCE) was introduced in API 1, they specify its value (as "android.resource"), and they don't mention it ever having changed. So this is specified to be a fixed constant.

If a future API version (or a rogue device vendor) wants to do something differently here, it's likely to involve different semantics that require code changes to adapt to anyway; in the unlikely event that they change the existing constant's value, it's not clear that we'd be closer to working by following the constant vs. following the value.

What using the constant *does* do for us is explicitly draw the connection with what bit of the system API we're relying on. Helps catch typos, and better yet makes it easier to look up relevant docs. This is a good example: I actually found it hard to google for information about these resource URIs (to answer my questions about `R.raw.zulip` vs. `"raw/zulip"`), but then I looked up [`ContentResolver.SCHEME_ANDROID_RESOURCE`](https://developer.android.com/reference/android/content/ContentResolver#SCHEME_ANDROID_RESOURCE), and the helpful docs I linked to above which answered my questions are there on the same page.